test/explorer/series/inferred_dtype_property_test.exs

```diff
@@ -0,0 +1,100 @@
+defmodule Explorer.Series.InferredDtypePropertyTest do
+  @moduledoc """
+  Property tests for checking the inferred dtype logic when the dtype isn't
+  specified in `Explorer.Series.from_list/1`.
+
+  ## Notes
+
+    * A maximum of 3 used quite a bit. This is intentional. Usually issues stem
+      from empty lists, not really long lists. By keeping lists small, we can
+      iterate much quicker through the input space.
+  """
+  use ExUnit.Case, async: true
+  use ExUnitProperties
+
+  alias Explorer.Series
+
+  @moduletag timeout: :infinity
+
+  property "inferred dtype should always be a sub-dtype" do
+    check all(
+            dtype <- dtype_generator(),
+            series <- series_of_dtype_generator(dtype),
+            max_run_time: 60_000,
+            max_runs: 10_000
+          ) do
+      assert series |> Series.dtype() |> sub_dtype_of?(dtype)
+    end
+  end
+
+  defp dtype_generator do
+    scalar_dtype_generator = StreamData.constant({:s, 64})
+
+    # We don't need complicated keys: single letter strings should suffice.
+    key_generator = StreamData.string(?a..?z, min_length: 1, max_length: 1)
+
+    dtype_generator =
+      StreamData.tree(scalar_dtype_generator, fn generator ->
```

Comment:
It is up to you, but maybe StreamData should be imported. :)

Reply:
Oh yeah that's more readable: [a78b283](https://github.com/elixir-explorer/explorer/pull/856/commits/a78b283acbc7bdf495e3ae13b287033a7cbe21db)